Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Cursors/BaseCursor.cs

```diff
@@ -113,6 +114,33 @@ public float LookRotationBlend
         [Tooltip("Blend value for surface normal to user facing lerp")]
         private float lookRotationBlend = 0.5f;
 
+        /// <summary>
+        /// Dictates whether the cursor should resize with distance or stay a constant size.
+        /// </summary>
+        public bool ResizeCursorWithDistance
+        {
+            get { return ResizeCursorWithDistance; }
+            set { resizeCursorWithDistance = value; }
+        }
+
+        [Header("Scaling")]
+        [SerializeField]
+        [Tooltip("Dictates whether the cursor should resize with distance (to always appear the same size) or stay a constant scale.")]
+        private bool resizeCursorWithDistance = true;
+
+        /// <summary>
+        /// Dictates whether the cursor should resize with distance or stay a constant size.
+        /// </summary>
+        public float CursorSizeAsPercentageOfViewport
+        {
+            get { return cursorSizeAsPercentageOfViewport; }
+            set { cursorSizeAsPercentageOfViewport = value; }
+        }
+
+        [SerializeField, Range(0f, 1f)]
+        [Tooltip("If resizeCursorWithDistance is enabled, this sets how large the cursor will become (as a percentage of the viewport).")]
+        private float cursorSizeAsPercentageOfViewport = 0.0125f;
```

Comment:
I'm not sure if this approach will work as expected with all the platforms we're supporting - we can't rely on them having the same / similar FOVs - Does this work well for both HL1 and HL2? How does this look on mobile?

Usually we're using cm / m as units for properties in our UX components. I was wondering if we could maybe have a size in cm or fixed scale applied to the cursor in 1m distance - this would be the value we set up and then we calculate the cursor scaling from this reference which should be the same on all the platforms? Would that make sense?

Reply:
Yes I think so! Currently, it _does_ look correctly on HL1/HL2 but when VR headsets are used, FOV of course is bigger. On the Odyssey, it doesnt look _too_ big, but definitely bigger. Thought about angular scale vs. percentage-of-viewport scale a lot yesterday evening and for the multi-platform reason alone I think I will switch the logic to be more like what you're describing (combination of fixed measurement at one meter deep and angular scale) which will lead to more platform independence.